packages/smooth_app/lib/pages/prices/infinite_scroll_controller.dart

```diff
@@ -0,0 +1,99 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+/// A generic controller for handling infinite scrolling in lists.
+/// [T] is the type of items being displayed.
+/// [P] is the type of parameters used for pagination.
+class InfiniteScrollController<T, P> {
+  InfiniteScrollController({
+    required this.fetchItems,
+    required Iterable<T> initialItems,
+  })  : _currentPage = _initialPage,
+        _items = List<T>.from(initialItems),
+        initialItems = List<T>.from(initialItems);
+
+  /// Returns a Future with the fetched items
```

Comment:
```suggestion
  /// Future with the fetched items
```
It's a field, it doesn't "return".